helm/app/templates/application/app/deployment.yaml

```diff
@@ -9,7 +9,6 @@ metadata:
     app.kubernetes.io/component: app
     app.service: {{ $.Values.resourcePrefix }}app
 spec:
-  replicas: 1
```

Comment:
Ideally this should be a parameter instead as HPA not always wanted. the PDB already checks for service.*.replicas setting so

```
    {{- with (pick . "replicas") }}
    {{- . | toYaml | nindent 2 }}
    {{- end }}
```

Reply:
The default is 1, though I can see the case of accidentally scaling to 0 and the service never coming back.

Reply:
I'm not talking about the deletion, just what it could be instead of a deletion. If .replicas isn't set it's the same as deletion, but allows for the case of fixed replica count without a HPA

Reply:
Updated with your suggestion (Co-Authored-By added)

Reply:
helm/argo would delete the replicas setting in this case of removal, so will fall back to the default of 1